src/control/client/pool.go

```diff
@@ -27,30 +27,83 @@ import (
 	"golang.org/x/net/context"
 
 	pb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
+	"github.com/daos-stack/daos/src/control/logging"
+	"github.com/pkg/errors"
 )
 
-// CreatePool will create a DAOS pool using provided parameters and return uuid
-func (c *connList) CreatePool(req *pb.CreatePoolReq) ResultMap {
-	results := make(ResultMap)
-	mc := c.controllers[0] // connect to first AP only for now
+func verifyUnicast(cs []Control) (Control, error) {
+	if len(cs) > 1 {
+		return nil, errors.Errorf("unexpected number of connections, "+
+			"want 1, have %d", len(cs))
+	}
 
-	resp, err := mc.getSvcClient().CreatePool(context.Background(), req)
+	return cs[0], nil
+}
+
+// PoolCreate will create a DAOS pool using provided parameters and return
+// uuid, list of service replicas and error (including any DER code from DAOS).
+//
+// Isolate protobuf encapsulation in client and don't expose to calling code.
+func (c *connList) PoolCreate(log *logging.LeveledLogger, scmBytes uint64,
```

Comment:
I would prefer to see these kinds of API parameters passed via request struct rather than function arguments. It makes the API surface clearer and allows for default values to be set. I realize that it means you are duplicating much of what is in the protobuf definition, but that's OK. The goal isn't to save lines of code, it's to provide a stable API surface that is distinct from the communications implementation. We could in theory swap out protobufs for something else and our API wouldn't have to change. I doubt that is likely to happen, but more to the point it means that API callers don't need to know anything about crafting protobuf requests or interpreting protobuf results.